### PR TITLE
Add 1s cache for lapi.GetStreamByPlaybackID()

### PR DESCRIPTION
**Pull request by leszko**
It causes the prod failure, because of the cascading failures

Related Discord thread: https://discord.com/channels/423160867534929930/1265176750544130068/1265181494763327562

### CODE REVIEW

File: mapic/utils.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package mistapiconnector
+
+import (
+	"github.com/livepeer/go-api-client"
+	"sync"
+	"time"
+)
+
+const lapiCacheDuration = 1 * time.Second
```
Comment (victorges) [inline mapic/utils.go:9]: Do we need the latest version of a stream for any config that can be changed? Maybe this cache could be even higher

File: mapic/utils.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	// Check again in case another goroutine has updated the cache in the meantime
+	e, ok = a.streamCache[playbackId]
+	if ok && e.stream != nil && e.updateAt.Add(a.ttl).After(time.Now()) {
+		return e.stream, e.err
+	}
+	// No value in the cache, fetch from Livepeer API and store the result in a cache
+	stream, err := a.lapi.GetStreamByPlaybackID(playbackId)
+	a.streamCache[playbackId] = entry{
+		stream:   stream,
+		err:      err,
```
Comment (victorges) [inline mapic/utils.go:54]: do we need to cache error cases if we only use cache when it's successful? Could be clearer to have an early return for `err != nil` here instead and skip the err in cache

**Review by victorges — APPROVED** [review]
LGTM

File: mapic/utils.go (hunk 1 above)
Comment (leszko) [reply mapic/utils.go:9]: Not sure. I'm just afraid to set it higher (and 1s is enough for what we encountered)

File: mapic/utils.go (hunk 2 above)
Comment (leszko) [reply mapic/utils.go:54]: We do want to cache errors as well. That was actually the root cause of the issue that we never cached failed requests so they piled up.

**Pull request merged**